backend/test/services/event/event_test.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for the EventService class."""
+
+# PyTest
+import pytest
+from unittest.mock import create_autospec
+
+from backend.services.event import EventNotFoundException
+from backend.services.exceptions import UserPermissionException
+
+# Tested Dependencies
+from ....models import Event, EventDetails
+from ....services import EventService
+
+# Injected Service Fixtures
+from ..fixtures import event_svc_integration
+
+# Explicitly import Data Fixture to load entities in database
+from ..core_data import setup_insert_data_fixture
+
+# Data Models for Fake Data Inserted in Setup
+from .event_test_data import (
+    events,
+    event_one,
+    to_add,
+    updated_event
+)
+from ..user_data import root, user
+
+# Test Functions
+
+
+def test_get_all(event_svc_integration: EventService):
+    """Test that all events can be retrieved."""
+    fetched_events = event_svc_integration.all()
+    assert fetched_events is not None
+    assert len(fetched_events) == len(events)
+    assert isinstance(fetched_events[0], EventDetails)
+
+
+def test_get_from_id(event_svc_integration: EventService):
+    """Test that events can be retrieved based on their ID."""
+    fetched_event = event_svc_integration.get_from_id(1)
+    assert fetched_event is not None
+    assert isinstance(fetched_event, Event)
+    assert fetched_event.id == event_one.id
+
+
+def test_create_enforces_permission(event_svc_integration: EventService):
+    """Test that the service enforces permissions when attempting to create an event."""
+
+    # Setup to test permission enforcement on the PermissionService.
+    event_svc_integration._permission = create_autospec(
+        event_svc_integration._permission
+    )
+
+    # Test permissions with root user (admin permission)
+    event_svc_integration.create(root, to_add)
+    event_svc_integration._permission.enforce.assert_called_with(
+        root, "organization.events.create", f"organization/{to_add.organization_id}"
+    )
+
+
+def test_create_event_as_root(event_svc_integration: EventService):
+    """Test that the root user is able to create new events."""
+    created_event = event_svc_integration.create(root, to_add)
+    assert created_event is not None
+    assert created_event.id is not None
+
+
+def test_create_event_as_user(event_svc_integration: EventService):
+    """Test that any user is *unable* to create new events."""
+    with pytest.raises(UserPermissionException):
+        event_svc_integration.create(user, to_add)
+        pytest.fail()  # Fail test if no error was thrown above
+
+
+def test_get_events_from_organization(event_svc_integration: EventService):
+    """Test that list of events can be retrieved based on specified organization."""
+    event_svc_integration.create(root, to_add)
```

Comment:
Small Nit / Discussion: Unsure if we should have a create call in here since this could be implemented incorrectly - but since the unit tests run sequentially, it may be okay.

Reply:
Yeah I figured since the test above tests creating an object it would be fine but I can refactor if necessary.

Reply:
@atoneyd no worries, I think this is okay!